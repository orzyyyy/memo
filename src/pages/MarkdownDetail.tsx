import React from 'react';
import './css/MarkdownDetail.css';
import 'github-markdown-css/github-markdown.css';

export interface MarkdownDetailProps {
  dataSource: string;
}

const MarkdownDetail = (props: MarkdownDetailProps) => (
  <div className="MarkdownDetail markdown-body">
    <div className="wrapper">
      <div className="content" dangerouslySetInnerHTML={{ __html: props.dataSource || '' }} />
    </div>
  </div>
);

export default MarkdownDetail;
