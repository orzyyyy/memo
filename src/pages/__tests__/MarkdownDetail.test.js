import React from 'react';
import { mount } from 'enzyme';
import MarkdownDetail from '../MarkdownDetail';
import 'nino-cli/scripts/setup';

describe('MarkdownDetail', () => {
  it('render correctly', () => {
    const wrapper = mount(<MarkdownDetail dataSource={'- test'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
