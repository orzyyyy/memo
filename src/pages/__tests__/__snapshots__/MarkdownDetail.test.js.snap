// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownDetail render correctly 1`] = `
<Component
  dataSource="- test"
>
  <div
    className="MarkdownDetail markdown-body"
  >
    <div
      className="wrapper"
    >
      <div
        className="content"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<ul>
<li>test</li>
</ul>
",
          }
        }
      />
    </div>
  </div>
</Component>
`;
